exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1550 @@
+	by.Websocket = stream.New()
+	by.WebsocketResponseMaxLimit = exchange.DefaultWebsocketResponseMaxLimit
+	by.WebsocketResponseCheckTimeout = exchange.DefaultWebsocketResponseCheckTimeout
+	by.WebsocketOrderbookBufferLimit = exchange.DefaultWebsocketOrderbookBufferLimit
+}
+
+// Setup takes in the supplied exchange configuration details and sets params
+func (by *Bybit) Setup(exch *config.Exchange) error {
+	if !exch.Enabled {
+		by.SetEnabled(false)
+		return nil
+	}
+
+	err := by.SetupDefaults(exch)
+	if err != nil {
+		return err
+	}
+
+	wsRunningEndpoint, err := by.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	if err != nil {
+		return err
+	}
+
+	err = by.Websocket.Setup(
+		&stream.WebsocketSetup{
+			ExchangeConfig:        exch,
+			DefaultURL:            bybitWSBaseURL + wsSpotPublicTopicV2,
+			RunningURL:            wsRunningEndpoint,
+			RunningURLAuth:        bybitWSBaseURL + wsSpotPrivate,
+			Connector:             by.WsConnect,
+			Subscriber:            by.Subscribe,
+			Unsubscriber:          by.Unsubscribe,
+			GenerateSubscriptions: by.GenerateDefaultSubscriptions,
+			Features:              &by.Features.Supports.WebsocketCapabilities,
+			OrderbookBufferConfig: buffer.Config{
+				SortBuffer:            true,
+				SortBufferByUpdateIDs: true,
+			},
+			TradeFeed: by.Features.Enabled.TradeFeed,
+		})
+	if err != nil {
+		return err
+	}
+
+	err = by.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  by.Websocket.GetWebsocketURL(),
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+	})
+	if err != nil {
+		return err
+	}
+
+	return by.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		URL:                  bybitWSBaseURL + wsSpotPrivate,
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		Authenticated:        true,
+	})
+}
+
+// AuthenticateWebsocket sends an authentication message to the websocket
+func (by *Bybit) AuthenticateWebsocket(ctx context.Context) error {
+	return by.WsAuth(ctx)
+}
+
+// Start starts the Bybit go routine
+func (by *Bybit) Start(wg *sync.WaitGroup) error {
+	if wg == nil {
+		return fmt.Errorf("%T %w", wg, common.ErrNilPointer)
+	}
+	wg.Add(1)
+	go func() {
+		by.Run()
+		wg.Done()
+	}()
+	return nil
+}
+
+// Run implements the Bybit wrapper
+func (by *Bybit) Run() {
+	if by.Verbose {
+		log.Debugf(log.ExchangeSys,
+			"%s Websocket: %s.",
+			by.Name,
+			common.IsEnabled(by.Websocket.IsEnabled()))
+		by.PrintEnabledPairs()
+	}
+
+	if !by.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+
+	err := by.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			by.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (by *Bybit) FetchTradablePairs(ctx context.Context, a asset.Item) ([]string, error) {
```

Comment:
Its hard to keep track of all the comments, but this one was recently added and I think it would be good to add/test:
https://github.com/thrasher-corp/gocryptotrader/pull/887#discussion_r881157099

From eyeballing, it looks like it matches what is part of here: https://www.bybit.com/data/markets/contract and https://www.bybit.com/data/markets/spot with exception to BTC-PERP